aten/src/THC/THCCachingAllocator.cpp

```diff
@@ -80,6 +80,12 @@ struct THCCachingAllocator
   typedef bool (*Comparison)(const Block*, const Block*);
   typedef std::set<Block*, Comparison> FreeBlocks;
 
+  // max memory allocated
+  static size_t max_memory_allocated;
+
+  // current memory allocated
+  static size_t memory_allocated;
```

Comment:
It's a bit counterintuitive that this is a sum over all GPUs. One is usually interested in per-GPU usage.

Reply:
Should the new methods instead give memory usage stats on a given GPU? I can make a map here to support that.

Reply:
Yes, that seems useful and strictly more general.